src/Router.js

```diff
@@ -1,16 +1,64 @@
+import React, { useState } from 'react';
 import { BrowserRouter, Routes, Route } from 'react-router-dom';
-import Main from 'pages/Main/Main';
+import { useFruitzState } from './FruitContext';
+import Main from './pages/Main/Main';
 import Login from './pages/Login/Login';
 import Join from './pages/Join/Join';
+import Fruit from './pages/Product/Fruit/Fruit';
+import Beverage from './pages/Product/Beverage/Beverage';
+import Detail from './pages/Product/Components/Detail/Detail';
+import Goods from './pages/Product/Goods/Goods';
+import Gifts from './pages/Product/Gifts/Gifts';
+import Aside from './components/Aside/Aside';
+import Header from './components/Header/Header';
+import Cart from './pages/Cart/Cart';
+import Contact from './pages/Contact/Contact';
+import Footer from './components/Footer/Footer';
+import MyPage from './pages/MyPage/MyPage';
+import ScrollToTop from './ScrollToTop';
 
 function Router() {
+  const token = localStorage.getItem('fruitz_user');
+  const [isLogin, setIsLogin] = useState(token ? true : false);
+  const productData = useFruitzState();
+
   return (
     <BrowserRouter>
+      <ScrollToTop />
+      <Header isLogin={isLogin} setIsLogin={setIsLogin} />
+      <Aside />
       <Routes>
-        <Route path="/" element={<Main />} />
-        <Route path="/member/login" element={<Login />} />
+        <Route path="/" element={<Main setIsLogin={setIsLogin} />} />
         <Route path="/member/join" element={<Join />} />
+        <Route
+          path="/member/login"
+          element={<Login setIsLogin={setIsLogin} />}
+        />
+        <Route path="/product/fruit" element={<Fruit />} />
+        <Route path="/product/beverage" element={<Beverage />} />
+        <Route path="/product/goods" element={<Goods />} />
+        <Route path="/product/gifts" element={<Gifts />} />
+        <Route
+          path="/product/fruit/detail/:id"
+          element={<Detail productData={productData.data} />}
+        />
+        <Route
+          path="/product/beverage/detail/:id"
+          element={<Detail productData={productData.data} />}
+        />
+        <Route
+          path="/product/goods/detail/:id"
+          element={<Detail productData={productData.data} />}
+        />
+        <Route
+          path="/product/gifts/detail/:id"
+          element={<Detail productData={productData.data} />}
+        />
```

Comment:
```suggestion
        <Route
          path="/:product/fruit/detail/:id"
          element={<Detail productData={productData.data} />}
        />
```

다 작성하는게 아니라 동적라우팅 활용하는게 좋아보이네요!

Reply:
동적라우팅으로 하였습니다. 추후에 쿼리스트링이나 더 나은방향으로 연구해보겠습니다.